### PR TITLE
DROOLS-1456: avoid NPE and NoSuchElementException when a function/dec…

### PR DESCRIPTION
**Pull request by etirelli**
* avoid NPE and NoSuchElementException when a function/decision table correctly returns null as result

* avoid NPE when a literal expression does not contain any expression

### CODE REVIEW

File: kie-dmn-feel/src/main/java/org/kie/dmn/feel/util/Either.java [hunk 1]
```diff
@@ -46,6 +46,6 @@ public boolean isRight() {
     
     public <X> X cata(Function<L,X> left, Function<R,X> right) {
         // warning: left.invoke, because of FEEL specs, could return null. The below is the safest way to implement cata over this Either.
-        return isLeft() ? left.apply( this.left.get() ) : right.apply( this.right.get() );
+        return isLeft() ? left.apply( this.left.get() ) : right.apply( this.right.orElse( null ) );
```
Comment (etirelli) [inline kie-dmn-feel/src/main/java/org/kie/dmn/feel/util/Either.java]: @tarilabs I am not sure if this is the best way to solve this problem, but the fact is that functions and decision tables can correctly return null as a result and we need to support this case.

**Comment by etirelli** [comment]
@tarilabs can you please review?

File: kie-dmn-feel/src/main/java/org/kie/dmn/feel/util/Either.java (hunk 1 above)
Comment (tarilabs) [inline kie-dmn-feel/src/main/java/org/kie/dmn/feel/util/Either.java]: Yep, can you only kindly add a comment to `Either.isRight()` just before merging, as a note for posterity that:
this implementation of Either is tailored to support FEEL spec as functions could explicitly return null, hence two definition of isRight are possible

- negated of `isLeft`, where `isLeft()` is defined as `left.isPresent()`
or
- `isRight` defined as `right.isPresent() || (!right.isPresent() && !left.isPresent())`

so we opted for the former as a matter of simplicity?

I don't want that later I may inadvertently touch this Either making troubles down the line :)

Comment (tarilabs) [inline kie-dmn-feel/src/main/java/org/kie/dmn/feel/util/Either.java]: Also, comment at line 48 is stating `left.invoke` but now I noticed that was affecting `BaseFEELFunction` in commit 0ba42e7 but here it's the JDK's Function so you can correct it back if you want

**Pull request merged**